src/plugins/intel_cpu/src/emitters/jit_snippets_emitters.cpp

```diff
@@ -0,0 +1,575 @@
+        std::tie(in_regs, out_regs) = c.second;
+        if (auto tile_scheduler = std::dynamic_pointer_cast<TileSchedulerEmitter>(emitter)) {
+            in_regs.push_back(static_cast<size_t>(dnnl::impl::cpu::x64::abi_param1.getIdx()));
+            in_regs.push_back(static_cast<size_t>(dnnl::impl::cpu::x64::abi_param2.getIdx()));
+            out_regs = gp_regs_used;
+        }
+        emitter->emit_code(in_regs, out_regs, vec_regs_pool, gp_regs_pool);
+    }
+    h->postamble();
+}
+
+TileSchedulerEmitter::TileSchedulerEmitter(dnnl::impl::cpu::x64::jit_generator* h, dnnl::impl::cpu::x64::cpu_isa_t isa,
+                                           const std::shared_ptr<ov::Node>& n) : jit_container_emitter(h, isa, n) {
+    const auto tile_scheduler = ov::as_type_ptr<ngraph::snippets::op::TileScheduler>(n);
+    if (!tile_scheduler)
+        IE_THROW() << "TileSchedulerEmitter invoked with invalid op argument";
+    if (!tile_scheduler->compile_params)
+        IE_THROW() << "TileEmitter invoked without compile_params";
+    body = {tile_scheduler->vector_region, tile_scheduler->scalar_region};
+    jcp = *reinterpret_cast<const jit_snippets_compile_args*>(tile_scheduler->compile_params);
+}
+void TileSchedulerEmitter::emit_code(const std::vector<size_t> &in,
+                                     const std::vector<size_t> &out,
+                                     const std::vector<size_t> &pool,
+                                     const std::vector<size_t> &gpr) const {
+    validate_arguments(in, out, pool, gpr);
+    emit_impl(in, out, pool, gpr, nullptr);
+}
+void TileSchedulerEmitter::validate_arguments(const std::vector<size_t> &in,
+                                     const std::vector<size_t> &out,
+                                     const std::vector<size_t> &pool,
+                                     const std::vector<size_t> &gpr) const {
+    if (in.size() != 5)
+        IE_THROW() << "TileSchedulerEmitter got invalid number of inputs. Expected 5, got " << in.size();
+    if (out.size() != in[0] + in[1])
+        IE_THROW() << "TileSchedulerEmitter got invalid number of outputs. Expected " << in[0] + in[1] << " , got " << out.size();
+    if (body.size() != 2)
+        IE_THROW() << "TileSchedulerEmitter got invalid body size, expected 2 (vector & scalar TileEmitter), got " << body.size();
+    if (!(std::dynamic_pointer_cast<TileEmitter>(body[0].first) && std::dynamic_pointer_cast<TileEmitter>(body[1].first)))
+        IE_THROW() << "TileSchedulerEmitter can contain only TileEmitters inside its body";
+}
+
+void TileSchedulerEmitter::emit_impl(const std::vector<size_t>& in,
+                                     const std::vector<size_t>& out,
+                                     const std::vector<size_t>& vec_pool,
+                                     const std::vector<size_t>& gpr_pool,
+                                     const ov::intel_cpu::emitter_context *emit_context) const {
+    const size_t num_inputs = in[0];
```

Comment:
Should we check that `in` has exactly 3 items (or more 3 items)?

Reply:
That's an interesting part) `in` is expected to have 5 items, see [this check](https://github.com/maojan19/openvino/blob/874ee3acda6f5b0d7b3f0448dbbd4ab86b24e7ac/src/plugins/intel_cpu/src/emitters/jit_snippets_emitters.cpp#L165-L166) please. First 3 values are provided by [generator](https://github.com/maojan19/openvino/blob/874ee3acda6f5b0d7b3f0448dbbd4ab86b24e7ac/src/common/snippets/src/generator.cpp#L84-L85) as usual, but another 2 are added by KernelEmitter (look [here](https://github.com/maojan19/openvino/blob/874ee3acda6f5b0d7b3f0448dbbd4ab86b24e7ac/src/plugins/intel_cpu/src/emitters/jit_snippets_emitters.cpp#L134-L138) for details). I understand that this might be not very clear at first glance, but we had to do so because generator has no idea about system-specific ABI. Anyway, let's discuss it offline.

Reply:
Yeah, I got it. I forget that we have method `validate_arguments`